packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -270,14 +273,13 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
 
   // Call user-defined onChange when the value actually changes
   React.useEffect(() => {
-    if (selectedIndex != undefined && selectedIndex >= 0) {
-      const value = options[selectedIndex]?.value;
-      if (value === valueProp) {
```

Comment:
Why removed the `value === valueProp` check?

Reply:
@mayank99 this is same as I had in multi select.
2nd use effect has the `old` value of selectedIndex!
Even though I thought if we use reducer, it should get the newest one :O

Reply:
If i remember correctly, this check was added because people were not always memoizing the onChange callback and it was leading to infinite loops 🤔 Should we use the same `queueMicrotask` logic here?

Reply:
Would be nicer if we could avoid `queueMicrotask`... Looks hacky.
Dont we add ref of `onChange` ourselves, so we dont care about users memoizing it...? Or its different

Reply:
Hmm yeah, I think `onChange` might be fine because of the ref but people also don't always memoize `options`. If you try to recreate #697 and #709 in playground, you will find that it still loops infinitely.

Reply:
I would leave `options` issue to other PR. This one fixes #697 so maybe merge?

Reply:
But does it?
@mayank99 is saying #697 is still reproducible with your fix?

Reply:
yeah #697 is reproducible with current changes if `options` is not memoized